compaction/compaction_manager.cc

```diff
@@ -412,15 +419,15 @@ class compaction_manager::custom_compaction_task : public compaction_manager::ta
     {}
 
 protected:
-    virtual future<> do_run() override {
+    virtual future<compaction_counters_opt> do_run() override {
```

Comment:
Why use optional here? Can't we just return default constructed stats struct instead?

Reply:
In any case this (adding the return value to `perform_task()` should be in a commit of its own.

Reply:
At first it returned default constructed but scrub specific struct and the change to optional was requested. Currently it is general for all compaction tasks though. Should it be changed to default constructed value? @bhalevy @denesb

Reply:
I don't have a strong opinion on this.

Reply:
Actually: with the current interface, one has to know which concrete task actually returns an engaged optional. With default-initialized stats struct, one can just use the counters returned, they will be 0 if not set. @bhalevy WDYT?